src/sentry/notifications/platform/target.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass, field
+from typing import Any
+
+from sentry.integrations.services.integration import integration_service
+from sentry.integrations.services.integration.model import (
+    RpcIntegration,
+    RpcOrganizationIntegration,
+)
+from sentry.notifications.platform.provider import NotificationProvider
+from sentry.notifications.platform.registry import provider_registry
+from sentry.notifications.platform.types import (
+    NotificationProviderKey,
+    NotificationTargetResourceType,
+)
+from sentry.utils.registry import NoRegistrationExistsError
+
+
+class NotificationTargetError(Exception):
+    pass
+
+
+@dataclass(kw_only=True)
```

Comment:
I'm a keyword only proponent anyway, but this is actually necessary in order to make a dataclass which extends this one (`IntegrationNotificationTarget`) with new required fields, but _after_ the optional field  of `specific_data`, just because of how python inheritance works.

Reply:
I've got no problem making this `kw_only`. We can always implement a different class that processes these in various ways if we need any sort of data manipulation. On that same note, shouldn't we also freeze this impl? Preventing code from modifying targets seems worthwhile to me.